### DISCUSSION TITLE
Is it possible for nvim-dap to debug a neovim plugin currently in use?

### DISCUSSION THREAD

**Discussion by marvindore** [mfussenegger/nvim-dap#782] [General]
Hello all,
I am currently using the Packer package manager and I have cloned the repository to a plugin locally and loaded it into Packer like `~/git/PluginRepo`. I am trying to track down a bug inside this plugin but want to know if nvim-dap can attach to this plugin some how?

**Comment by mfussenegger** [comment]
You can debug neovim-lua with https://github.com/jbyuki/one-small-step-for-vimkind, but you'll need 2 neovim instances, one where you set the breakpoints and step through the code, and one where you use the plugin you want to debug, to trigger the breakpoints. Once a breakpoint is hit, the other instance will lock/freeze until you step around/continue.

I have a special adapter configuration that spawns a new neovim instance in a new terminal to make the workflow a bit eaiser.

Short-demo:

https://user-images.githubusercontent.com/38700/207815884-1eb670b5-6ef4-42ee-b5f7-8c277d896ddf.mp4

Configuration:

https://github.com/mfussenegger/dotfiles/blob/2428fc92505bb5aa5c48d475f1f7a23cbf64fc00/vim/.config/nvim/ftplugin/lua.lua#L3-L43
https://github.com/mfussenegger/dotfiles/blob/2428fc92505bb5aa5c48d475f1f7a23cbf64fc00/vim/.config/nvim/ftplugin/lua.lua#L56-L102

**Reply by marvindore** [reply]
Hi @mfussenegger , Thank you so much for this information as I am new to neovim & Lua, I was stuck on this question for some time. I will give this a shot!
